src/main/java/org/tdl/vireo/model/User.java

```diff
@@ -457,6 +457,19 @@ public void loadActiveFilter(NamedSearchFilterGroup filter) {
     public void setFilterColumns(List<SubmissionListColumn> filterColumns) {
         this.filterColumns = filterColumns;
     }
+    
+    public void addFilterColumn(SubmissionListColumn submissionListColumn) {
+    	this.filterColumns.add(submissionListColumn);
+    }
+    
+    public void removeFilterColumn(SubmissionListColumn submissionListColumn) {
+    	this.filterColumns.forEach(fc->{
+    		if(fc.getId() == submissionListColumn.getId()) {
+    			int indexOfFc = this.filterColumns.indexOf(fc);
+    			this.filterColumns.remove(indexOfFc);
```

Comment:
Not safe to remove from a collection while iterating over it. Use a iterator and a while loop or set the item aside to remove or use a traditional for loop with a break.